qa/sc_evm_native_interop.py

```diff
@@ -157,6 +174,163 @@ def run_test(self):
         # Verify gas usage of the nested call to the native contract reported by the trace matches with the estimation
         assert_equal(int(native_call["gasUsed"], 16) + intrinsic_gas, int(estimation_native["result"], 16))
 
+        """
+        Tests from Native Smart contract to EVM Smart contract
+        """
+        # Compile and deploy the Storage contract
+        initial_value = 142
+        storage_contract, storage_contract_address = self.deploy("Storage", initial_value)
+
+        method_inc = 'inc()'
+        method_retrieve = 'retrieve()'
+
+        sol_contract_call_data_inc = storage_contract.raw_encode_call(method_inc)
+        sol_contract_call_data_retrieve = storage_contract.raw_encode_call(method_retrieve)
+
+        # Get current value in Storage by calling the EVM contract directly
+        expected_value = node.rpc_eth_call(
+            {
+                "to": storage_contract_address,
+                "input": sol_contract_call_data_retrieve
+            }, "latest"
+        )
+        value = int(expected_value['result'], 16)
+        assert_equal(initial_value, value)
+
+        # Get current value in Storage by calling the proxy native contract
+
+        method = 'invokeCall(address,bytes)'
+        abi_str = function_signature_to_4byte_selector(method)
+        encoded_abi_method_signature = encode_hex(abi_str)
+        addr_padded_str = "000000000000000000000000" + remove_0x_prefix(storage_contract_address)
+        data_input = encoded_abi_method_signature + addr_padded_str
+        data_input += "0000000000000000000000000000000000000000000000000000000000000040"
+        h_len = hex(len(remove_0x_prefix(sol_contract_call_data_retrieve)) // 2)
+        data_input += "000000000000000000000000000000000000000000000000000000000000000" + remove_0x_prefix(HexStr(h_len))
+        data_input += remove_0x_prefix(sol_contract_call_data_retrieve)
+        data_input += "00000000000000000000000000000000000000000000000000000000"
+
+        native_contract_address = format_evm(PROXY_SMART_CONTRACT_ADDRESS)
+        expected_value = node.rpc_eth_call(
+            {
+                "to": native_contract_address,
+                "input": data_input
+            }, "latest"
+        )
+
+        value = int(expected_value['result'], 16)
+        assert_equal(initial_value, value)
+
+        # Verify tracing
+        trace_response = node.rpc_debug_traceCall(
```

Comment:
I think we need to test the tracer call also with the other built-in tracers, at least the opcode logger and the 4byteTracer just to check that everything works correctly.

Reply:
Good idea. Done.